src/client/containers/UserProfilePage/UserProfilePage.Container.js

```diff
@@ -1,11 +1,23 @@
 import React from 'react';
 import './UserProfilePage.styles.css';
 import UserProfile from '../../components/UserProfile/UserProfile.component';
+import OrdersSummary from '../../components/OrdersSummary/OrderComponent';
+import { useAuthentication } from '../../hooks/useAuthentication';
+import Loader from '../../components/Loader/Loader.component';
 
 export default function UserProfilePage() {
+  const { user, isLoading } = useAuthentication();
+  const orders = (userId) => fetch(`api/orders/${userId}`);
+  const orderDetails = (userId) => fetch(`api/orders/${userId}`);
+
+  if (isLoading) {
+    return <Loader />;
+  }
+  
   return (
     <div>
       <UserProfile />
+      <OrdersSummary orders={orders(user.uid)} orderDetails={orderDetails(user.uid)}/>
```

Comment:
I don't see how this would work.
Both `orders` and `orderDetails` return Promises. There is no code here that either adds a `.then` handler or uses `await` to resolve those Promises. Later you're calling `.map` on a Promise which won't work.

Reply:
Hi Milton i tried a lot ,i will see after merge this, how it actually work ... i will fix changes later .